File: android/api-tests/src/test/java/com/revenuecat/apitests/java/StoreProductMapperApiTests.java
package com.revenuecat.apitests.java.mappers;

import com.revenuecat.purchases.hybridcommon.mappers.StoreProductMapperKt;
import com.revenuecat.purchases.models.StoreProduct;

import java.util.List;
import java.util.Map;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

@SuppressWarnings("unused")
class StoreProductMapperApiTests {
    private void checkItemMap(StoreProduct product) {
        Map<String, Object> map = StoreProductMapperKt.map(product);
    }

    private void checkListMapAsync(List<StoreProduct> products, Function1<? super List<? extends Map<String, ?>>, Unit> callback) {
        StoreProductMapperKt.mapAsync(products, callback);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/mappers/StoreProductMapperApiTests.kt
package com.revenuecat.apitests.kotlin.mappers

import com.revenuecat.purchases.hybridcommon.mappers.map
import com.revenuecat.purchases.hybridcommon.mappers.mapAsync
import com.revenuecat.purchases.models.StoreProduct

@Suppress("unused", "UNUSED_VARIABLE")
private class StoreProductMapperApiTests {
    fun checkItemMap(product: StoreProduct) {
        val map: Map<String, Any?> = product.map()
    }

    fun checkListMapAsync(products: List<StoreProduct>, callback: (List<Map<String, Any?>>) -> Unit) {
        val unit: Unit = products.mapAsync(callback)
    }
}
